10.8 Release Notes (#3740)
owncloud/docs: modules/admin_manual/pages/release_notes.adoc

* xref:changes-in-8-0[Changes in 8.0]
* xref:changes-in-7-0[Changes in 7.0]

== Changes in 10.8.0

Dear ownCloud administrator, please find below the changes and known issues in ownCloud Server 10.8 that need your attention. You can also read {owncloud-changelog-url}[the full ownCloud Server changelog] for further details on what has changed.

=== ownCloud Web supplements the Classic Web Interface - Try it!

> The all-new web interface for ownCloud, ownCloud Web, meanwhile has come a long way since its initial release end of 2020. It is available as an app on the https://marketplace.owncloud.com/apps/web[ownCloud Marketplace] and ownCloud Server has been xref:release_notes.adoc#owncloud-web-the-new-web-frontend-for-owncloud[prepared to work with it since version 10.6]. ownCloud Web https://owncloud.dev/clients/web/deployments/oc10-app/#configure-owncloud-10[can be deployed as a supplement to the classic web interface] and meanwhile it is in use at quite a number of installations. This has brought up good feedback around the integration with ownCloud 10 that has been addressed for 10.8. Additionally, lots of improvements have made their way into ownCloud Web. For an overview you can have a look at the https://owncloud.dev/ocis/release_notes/[ownCloud Infinite Scale release notes] and for a full list of changes, please consider the https://github.com/owncloud/web/blob/master/CHANGELOG.md[ownCloud Web changelog]. 

Reviewer: ```suggestion
The all-new web interface for ownCloud, ownCloud Web, has come a long way since its initial release at the end of 2020. It is available as an app on the https://marketplace.owncloud.com/apps/web[ownCloud Marketplace] and ownCloud Server has been xref:release_notes.adoc#owncloud-web-the-new-web-frontend-for-owncloud[prepared to work with it since version 10.6]. ownCloud Web https://owncloud.dev/clients/web/deployments/oc10-app/#configure-owncloud-10[can be deployed as a supplement to the classic web interface] and meanwhile it is in use at quite a number of installations. This has brought up good feedback around the integration with ownCloud 10 that has been addressed for 10.8. Additionally, lots of improvements have made their way into ownCloud Web. For an overview you can have a look at the https://owncloud.dev/ocis/release_notes/[ownCloud Infinite Scale release notes] and for a full list of changes, please see the https://github.com/owncloud/web/blob/master/CHANGELOG.md[ownCloud Web changelog]. 
```

The most prominent recent improvements are

* Accessibility: Compliance with WCAG 2.1 / BITV 2.0 (currently in the certification process)
* Theming: Ability to change name, logos, colors and more via config file (see the https://owncloud.dev/clients/web/theming/[documentation] for more information)
* Performance improvements in many areas

> With all these improvements we want to motivate ownCloud service providers to make ownCloud Web available to users, to experience the new technology as well as to provide feedback to further improve it. The ownCloud Web integration app comes as part of the 10.8 complete bundle. Furthermore, ownCloud Web has a release cycle of 3 weeks that is independent of ownCloud Server. This allows to get new versions with features, fixes and improvements regularly via the ownCloud Marketplace and to deploy them with minimal effort.

Reviewer: ```suggestion
With all these improvements we want to motivate ownCloud service providers to make ownCloud Web available to users, to experience the new technology as well as to provide feedback to further improve it. The ownCloud Web integration app comes as part of the 10.8 complete bundle. Furthermore, ownCloud Web has a release cycle of 3 weeks that is independent of ownCloud Server. This allows new versions with features, fixes and improvements to be made available regularly via the ownCloud Marketplace and to deploy them with minimal effort.
```

==== Fixed known issues

Both xref:release_notes.adoc#known-issues[known issues from ownCloud Server 10.7] have been fixed. The deployment of ownCloud Web is now more robust and administrators can optionally decide whether ownCloud Links (public and private links) should be provided by the Classic web interface or by ownCloud Web using a https://owncloud.dev/clients/web/deployments/oc10-app/#configure-link-routing[new option in config.php]. 

==== Feedback
As mentioned, features are still being added to ownCloud Web and the new web interface can't yet cover every use case of the Classic interface. To further shape the new product and to determine the development priorities it is of utmost importance to consider user feedback. We're very grateful for any hints or feedback that you supply via the following channels

> - Survey: LINK

Reviewer 2: Should `LINK` be replaced by a link or is the term valid

Author: Yes, placeholder for now.

- GitHub: https://github.com/owncloud/web[owncloud/web]
- Chat: talk.owncloud.com / #web

With one of the next ownCloud Web releases, there will be an integrated feedback feature that will allow users to directly report about their experience.

=== Improved Usability for "Add to ownCloud" on Public Links

> The usability of the "Add to ownCloud" feature on public link pages has been improved (https://github.com/owncloud/core/pull/38712[#38712]). Research on the feature has shown that most users use the feature to add links to the same ownCloud installation they originate from instead of creating federated shares with other ownCloud installations. For this reason there is a new default behavior which allows to add a link to a user account on the ownCloud installation the public link originates from with just a click and without requiring any further input. Advanced users can use a dropdown menu to add a link to a user account on another ownCloud installation.

Reviewer: ```suggestion
The usability of the "Add to ownCloud" feature on public link pages has been improved (https://github.com/owncloud/core/pull/38712[#38712]). Research on the feature has shown that most users use the feature to add links to the same ownCloud installation they originate from instead of creating federated shares with other ownCloud installations. For this reason there is a new default behavior which allows adding a link to a user account on the ownCloud installation the public link originates from with just a click and without requiring any further input. Advanced users can use a dropdown menu to add a link to a user account on another ownCloud installation.
```

=== Modernized Login Page

> The login page has received several improvements in regards of design and usability (https://github.com/owncloud/core/pull/38506[#38506]). A login button and labels for the input fields have been added and the "Alternative login" buttons (e.g., for authentication with OpenID Connect) have been redesigned to give a more modern look and feel. In this context, also the ownCloud Enterprise Theme has received improvements that are shipped with Server 10.8.

Reviewer: ```suggestion
The login page has received several improvements regarding design and usability (https://github.com/owncloud/core/pull/38506[#38506]). A login button and labels for the input fields have been added and the "Alternative login" buttons (e.g., for authentication with OpenID Connect) have been redesigned to give a more modern look and feel. In this context, also the ownCloud Enterprise Theme has received improvements that are shipped with Server 10.8.
```

=== Improvements for External Storages

* Performance and caching improvements for external storages (e.g., Windows Network Drives) have been added https://github.com/owncloud/core/pull/38804[#38804]
* Renaming or deleting a Windows Network Drive mount point from the files list is now properly prevented https://github.com/owncloud/core/pull/38709[#38709] https://github.com/owncloud/core/pull/38819[#38819]

> * Passwords in the external storage configuration can now be stored encoded in the database (e.g., for Windows Network Drive mount points). With the next release of the Windows Network Drive app, these can also be used to set up the WND Listener. https://github.com/owncloud/core/pull/38728[#38728]

Reviewer 2: `configuration can now be stored` or `configuration is now stored` ?

Author: "can now", as it depends on the external storage apps like WND to actually do it.

* The config report and the `occ user:report` command now provide the number of guest users https://github.com/owncloud/core/pull/38742[#38742] https://github.com/owncloud/configreport/pull/146[#146]
* For more control over Deleted Files, administrators can now decide if a resource should be deleted immediately instead of moving it to the trash bin. The behavior can be configured based on file extensions, directory names and size using the `trashbin_skip` options in config.php. https://github.com/owncloud/core/pull/38704[#38704]
* License keys can now be removed with a button in the admin settings https://github.com/owncloud/core/issues/38843[#38843]
* Video playback in ownCloud relies on browser capabilities. If a video cannot be played, users will now see a hint with guidance. https://github.com/owncloud/enterprise/issues/4632[#4632]

=== For developers

* App developers can now exclude default file actions like "Rename" when their app adds a new file view to ownCloud. https://github.com/owncloud/core/pull/38643[#38643]

=== Solved known issues

All xref:release_notes.adoc#known-issues[known issues from Server 10.7] have been fixed.

=== Known issues

> Currently there are no known issues with ownCloud Server 10.8. This section will be updated when more issues are discovered.

Reviewer: ```suggestion
Currently there are no known issues with ownCloud Server 10.8. This section will be updated if more issues are discovered.
```

There won't be any issues discovered ;)
